src/vs/workbench/services/languageRuntime/common/positronIPyWidgetsWebviewMessages.ts

```diff
@@ -66,7 +73,58 @@ export interface ICommOpenToWebview {
 	metadata: unknown;
 }
 
+export interface IGetPreferredRendererResultToWebview {
+	type: 'get_preferred_renderer_result';
+	parent_id: string;
+	/** The preferred renderer ID, or undefined if none found. */
+	renderer_id?: string;
+}
+
+export interface IKernelMessageClearOutput {
+	type: 'clear_output';
+	wait: boolean;
+}
+
+export interface IKernelMessageDisplayData {
+	type: 'display_data';
+	data: unknown;
+	metadata: unknown;
+}
+
+export interface IKernelMessageError {
+	type: 'error';
+	name: string;
+	message: string;
+	traceback: Array<string>;
+}
+
+export interface IKernelMessageExecuteResult {
+	type: 'execute_result';
+	data: unknown;
+	metadata: unknown;
+}
+
+export interface IKernelMessageStream {
+	type: 'stream';
+	name: 'stdout' | 'stderr';
+	text: string;
+}
```

Comment:
I think it would help me when looking back at this code if there was comments explaining when or why a message type would be sent. Just a sentence or two.